File: typescript/tsfind/src/findconfig.ts
/*
 * findconfig.ts
 *
 * Configuration values
 */

'use strict';

import * as path from 'path';

const isWin: boolean = /^win/.test(process.platform);

const HOME_NAME: string = isWin ? 'USERPROFILE' : 'HOME';
export const HOME: string = process.env[HOME_NAME] || '';

export class FindConfig {
  xfindPath: string;
  fileTypesPath: string;
  findOptionsPath: string;
  defaultFindSettingsPath: string;

  constructor() {
    const defaultXFindConfigDir = path.join(HOME, '.config', 'xfind');
    const xfindConfigDir = process.env.XFIND_CONFIG_DIR
      ? process.env.XFIND_CONFIG_DIR
      : defaultXFindConfigDir;
    const defaultXFindPath = path.join(HOME, 'src', 'xfind');
    const xfindPath = process.env.XFIND_PATH ? process.env.XFIND_PATH : defaultXFindPath;
    const jsfindPath = path.join(xfindPath, 'javascript', 'jsfind');
    const dataPath = path.join(jsfindPath, 'data');
    const fileTypesPath = path.join(dataPath, 'filetypes.json');
    const findOptionsPath = path.join(dataPath, 'findoptions.json');
    const defaultFindSettingsPath = path.join(xfindConfigDir, 'settings.json');

    this.xfindPath = xfindPath;
    this.fileTypesPath = fileTypesPath;
    this.findOptionsPath = findOptionsPath;
    this.defaultFindSettingsPath = defaultFindSettingsPath;
  }
}

File: ruby/rbfind/lib/rbfind/findconfig.rb
# frozen_string_literal: true

module RbFind

  # FindConfig - basic config info
  class FindConfig
    attr_reader :file_types_path
    attr_reader :find_options_path
    attr_reader :default_find_settings_path

    def initialize
      @file_types_path = File.realpath(File.join(File.dirname(__FILE__), "../../data/filetypes.json"))
      @find_options_path = File.realpath(File.join(File.dirname(__FILE__), "../../data/findoptions.json"))
      default_xfind_config_dir = File.join(ENV['HOME'], '.config', 'xfind')
      xfind_config_dir = ENV.fetch('XFIND_CONFIG_DIR', default_xfind_config_dir)
      @default_find_settings_path = File.join(xfind_config_dir, 'settings.json')
    end
  end
end
